### Setting the Stage -  Importing necessary libraries, and setting the path(s) needed to build the model!

In [50]:
import os
import sys
sys.path.append('/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages')
import pandas as pd
import matplotlib.pyplot as plt


### Understanding the Input - Reading and interpreting the basic details of our dataset 

In [51]:
column_names = ['sepal-length', 'sepal-width', 'petal-length', 'petal-width', 'class']
input_data = pd.read_csv('iris.data', names=column_names)

In [75]:
# Shape of input data
input_data.shape

(150, 5)

In [9]:
# First look at the dataset - to see how it looks!
input_data.head()

sepal-length  sepal-width  petal-length  petal-width        class
0           5.1          3.5           1.4          0.2  Iris-setosa
1           4.9          3.0           1.4          0.2  Iris-setosa
2           4.7          3.2           1.3          0.2  Iris-setosa
3           4.6          3.1           1.5          0.2  Iris-setosa
4           5.0          3.6           1.4          0.2  Iris-setosa

In [53]:
# Type of classes available (useful in a classification problem)
input_data["class"].unique()

array(['Iris-setosa', 'Iris-versicolor', 'Iris-virginica'], dtype=object)

In [80]:
# Number of instances of each class
input_data.groupby('class').size()

class
Iris-setosa        50
Iris-versicolor    50
Iris-virginica     50
dtype: int64

In [10]:
# A more statistical peep into the dataset
input_data.describe()

sepal-length  sepal-width  petal-length  petal-width
count    150.000000   150.000000    150.000000   150.000000
mean       5.843333     3.054000      3.758667     1.198667
std        0.828066     0.433594      1.764420     0.763161
min        4.300000     2.000000      1.000000     0.100000
25%        5.100000     2.800000      1.600000     0.300000
50%        5.800000     3.000000      4.350000     1.300000
75%        6.400000     3.300000      5.100000     1.800000
max        7.900000     4.400000      6.900000     2.500000

<font color='brown'>
    Some insights from the input_data that we have

* There are 3 classes of Iris flowers - Iris-setosa, Iris-versicolor, Iris-virginica
* There are a total of 150 rows in our dataset 
* Sepal details: <br>
    &emsp; Mean-length: 5.8 <br>
    &emsp; Mean-width:  3.05
* Petal details: <br>
    &emsp; Mean-length: 3.75 <br>
    &emsp; Mean-width:  1.19



</font>

### Visualizing the data - Plotting diagrams and graphs with the data

In [133]:
# Plotting box plot for the features
test = input_data[["sepal-length", "sepal-width", "petal-length", "petal-width"]]


<font color='brown'>
    Box-plots could tell us 5 important aspects of data: <br>
        &emsp;* Minimum value <br>
        &emsp;* First quartile(Q1)  <br>
        &emsp;* Median (second quartile (Q2)) <br>
        &emsp;* Third quartile (Q3)<br>
        &emsp;* Maximum 
    </font>